controllers/users.js

```diff
@@ -362,18 +362,25 @@ const getSelfDetails = async (req, res) => {
 const updateSelf = async (req, res) => {
   try {
     const { id: userId } = req.userData;
+    const { user } = await dataAccess.retrieveUsers({ id: userId });
+
     if (req.body.username) {
-      const { user } = await dataAccess.retrieveUsers({ id: userId });
       if (!user.incompleteUserDetails) {
         return res.boom.forbidden("Cannot update username again");
       }
       await userQuery.setIncompleteUserDetails(userId);
     }
 
-    const user = await userQuery.addOrUpdate(req.body, userId);
+    if (req.body.roles) {
+      if (user && user.roles.in_discord) {
+        return res.boom.forbidden("Cannot update roles");
+      }
```

Comment:
you have removed the `in_discord` in the validator file then it will not be there in the body Right ? The validator will throw an error of unknown key i.e Bad Request. And the code will not reach here. Correct me If I am wrong ?

Reply:
Here this case is only arise when user sign-in with GitHub and not joined discord yet so we can allow user who may join as maven, designer, product manager